File: backend/coffeeing/src/main/java/com/ssafy/coffeeing/modules/product/domain/Coffee.java
package com.ssafy.coffeeing.modules.product.domain;

import com.ssafy.coffeeing.modules.global.embedded.CoffeeCriteria;
import com.ssafy.coffeeing.modules.util.base.BaseEntity;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

import javax.persistence.AttributeOverride;
import javax.persistence.Column;
import javax.persistence.Embedded;
import javax.persistence.Entity;

@Getter
@SuperBuilder
@AllArgsConstructor(access = AccessLevel.PROTECTED)
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AttributeOverride(name = "id", column = @Column(name = "coffee_id"))
@Entity
public class Coffee extends BaseEntity {

    @Column
    private String regionKr;

    @Column
    private String regionEng;

    @Column
    private String coffeeNameKr;

    @Column
    private String coffeeNameEng;

    @Embedded
    private CoffeeCriteria coffeeCriteria;

    @Column
    private String flavorNote;
    
    @Column(columnDefinition = "text")
    private String imageUrl;

    @Column(columnDefinition = "text")
    private String productDescription;

    @Column
    @Builder.Default
    private Double totalScore = 0.0;

    @Column
    @Builder.Default
    private Integer totalReviewer = 0;

    @Column
    @Builder.Default
    private Integer popularity = 0;

    public void addReview(Integer score) {
        this.totalScore += score;
        this.totalReviewer++;
    }

    public void deleteReview(Double score) {
        this.totalScore -= score;
        this.totalReviewer--;
    }

    public void editReview(Integer diff) {
        this.totalScore += diff;
    }
}

File: model/coffeeing/app/database/models/coffee.py
from sqlalchemy import Column, ForeignKey
from sqlalchemy import BIGINT, Integer, String, DateTime, Double, Text
from sqlalchemy.orm import relationship

from ..connection import Base

class Coffee(Base):
    def __getitem__(self, key):
        return getattr(self, key)

    def __setitem__(self, key, value):
        return setattr(self, key, value)

    def keys(self):
        return list(self.__dict__.keys())[1:]

    def values(self):
        return list(self.__dict__.values())[1:]

    __tablename__ = "coffee"

    coffee_id = Column(BIGINT, primary_key=True, index=True)
    created_at = Column(DateTime())
    modified_at = Column(DateTime())
    acidity = Column(Double)
    body = Column(Double)
    roast = Column(Double)
    coffee_name_eng = Column(String(255))
    coffee_name_kr = Column(String(255))
    flavor_note = Column(String(255))
    image_url = Column(Text)
    product_description = Column(Text)
    region_eng = Column(String(255))
    region_kr = Column(String(255))
    total_reviewer = Column(Integer)
    total_score = Column(Integer)
    popularity = Column(Integer)